server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsAggregatorTests.java

```diff
@@ -0,0 +1,192 @@
+            aggregation -> aggregation.field(LONG_FIELD).includeExclude(includeExclude),
+            agg -> fail("test should have failed with exception"), ValueType.NUMERIC
+        ));
+        assertThat(e.getMessage(), equalTo("Aggregation [_name] cannot support regular expression style include/exclude " +
+            "settings as they can only be applied to string fields. Use an array of numeric values for include/exclude " +
+            "clauses used to filter numeric fields"));
+    }
+
+    private void testSearchCase(Query query, List<Long> dataset,
+                                Consumer<TermsAggregationBuilder> configure,
+                                Consumer<InternalMappedTerms> verify, ValueType valueType) throws IOException {
+        executeTestCase(false, query, dataset, configure, verify, valueType);
+    }
+
+    private void testSearchAndReduceCase(Query query, List<Long> dataset,
+                                         Consumer<TermsAggregationBuilder> configure,
+                                         Consumer<InternalMappedTerms> verify, ValueType valueType) throws IOException {
+        executeTestCase(true, query, dataset, configure, verify, valueType);
+    }
+
+    private void testBothCases(Query query, List<Long> dataset,
+                               Consumer<TermsAggregationBuilder> configure,
+                               Consumer<InternalMappedTerms> verify, ValueType valueType) throws IOException {
+        testSearchCase(query, dataset, configure, verify, valueType);
+        testSearchAndReduceCase(query, dataset, configure, verify, valueType);
+    }
+
+    private void executeTestCase(boolean reduced, Query query, List<Long> dataset,
+                                 Consumer<TermsAggregationBuilder> configure,
+                                 Consumer<InternalMappedTerms> verify, ValueType valueType) throws IOException {
+
+        try (Directory directory = newDirectory()) {
+            try (RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory)) {
+                Document document = new Document();
+                for (Long value : dataset) {
+                    if (frequently()) {
+                        indexWriter.commit();
+                    }
+
+                    document.add(new SortedNumericDocValuesField(LONG_FIELD, value));
+                    document.add(new LongPoint(LONG_FIELD, value));
+                    document.add(new SortedSetDocValuesField(KEYWORD_FIELD, new BytesRef(Long.toString(value))));
+                    indexWriter.addDocument(document);
+                    document.clear();
+                }
+            }
+
+            try (IndexReader indexReader = DirectoryReader.open(directory)) {
+                IndexSearcher indexSearcher = newIndexSearcher(indexReader);
+
+                TermsAggregationBuilder aggregationBuilder = new TermsAggregationBuilder("_name");
+                aggregationBuilder.userValueTypeHint(valueType);
```

Comment:
I'm concerned that all these tests are running with `userValueTypeHint` set.  We don't make it a required field in the parser, and I suspect most folks don't specify it.  We don't even specify it in the docs most of the time.  We should either test both with it set and not set, or default to not setting it.